middleware/http/opa/middleware.go

```diff
@@ -0,0 +1,196 @@
+package opa
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/dapr/components-contrib/middleware"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/valyala/fasthttp"
+)
+
+type middlewareMetadata struct {
+	Rego            string `json:"rego"`
+	DefaultStatus   int    `json:"defaultStatus,omitempty"`
+	IncludedHeaders string `json:"includedHeaders,omitempty"`
+}
+
+// NewMiddleware returns a new Open Policy Agent middleware
+func NewMiddleware(logger logger.Logger) *Middleware {
+	return &Middleware{logger: logger}
+}
+
+// Middleware is an OPA  middleware
+type Middleware struct {
+	logger logger.Logger
+}
+
+// RegoResult is the expected result from rego policy
+type RegoResult struct {
+	Allow             bool              `json:"allow"`
+	AdditionalHeaders map[string]string `json:"additional_headers,omitempty"`
+	StatusCode        int               `json:"status_code,omitempty"`
+}
+
+const (
+	opaErrorHeaderKey         = "x-dapr-opa-error"
+	opaErrorNoResult          = "received no results back from rego policy. Are you setting data.http.allow?"
```

Comment:
you can call errors.New() here and avoid calling when use. I don't think errors.New() decorate things with stacktrace, so there is no need to create new one every time. 

If you do so, it is customary to name an error like `errFoo`.

Reply:
Awesome! I wasn't sure, will fix that up now.

Reply:
Fixed up